src/main/scala/com/github/sbt/git/GitPlugin.scala

```diff
@@ -317,7 +318,12 @@ object SbtGit {
         } yield version
 
       // NOTE - Selecting the last tag or the first tag should be an option.
-      val highestVersion = versions.sortWith { versionsort.VersionHelper.compare(_, _) > 0 }.headOption
+      val highestVersion =
+        versions
+          .map(Version.parse)
+          .sortWith(_.compareTo(_) > 0)
+          .headOption
```

Comment:
sbt has https://github.com/sbt/librarymanagement/blob/develop/core/src/test/scala/sbt/librarymanagement/SemanticSelectorSpec.scala, which I think can be used for sorting with a few more lines of code.